### PR TITLE
Restify /contents API

### PR DESCRIPTION
**Pull request by gmelodie**
Addresses #583

## Verification
1. Add CAR, add to bucket (new name for collections) and dir too
image
1.2. List that bucket on that dir to see the file
image

2. Add CID, add to bucket and dir too
image
2.2. List that bucket on that dir to see the pinned file
image

3. Getting the status of one of the contents
image

4. Get content by CID
image

### CODE REVIEW

**Comment by kelindi** [comment]
@gmelodie Could we have the car files be passed through formData rather than through the body to be more consistent with how regular files are added

File: handlers.go [hunk 1]
```diff
@@ -157,32 +155,24 @@ func (s *Server) ServeAPI() error {
...
-	content.GET("/bw-usage/:content", withUser(s.handleGetContentBandwidth))
-	content.GET("/staging-zones", withUser(s.handleGetStagingZoneForUser))
-	content.GET("/aggregated/:content", withUser(s.handleGetAggregatedForContent))
-	content.GET("/all-deals", withUser(s.handleGetAllDealsForUser))
+	// to upload contents you only need an upload key
+	// to see info about contents you need a user-level key (see contents group)
+	e.POST("/contents", withUser(s.handleAdd), s.AuthRequired(util.PermLevelUpload))
+
+	contents := e.Group("", s.AuthRequired(util.PermLevelUser))
+	contents.GET("/contents", withUser(s.handleListContent))
```
Comment (anjor) [inline handlers.go:163]: Shouldn't these be behind /api/v2 ?

**Review by anjor — CHANGES_REQUESTED** [review]
We should not delete the v1 endpoints. They should still be around. The new endpoints should be under a v2 path.

Separately, I think uploadType should be an enum

**Comment by gmelodie** [comment]
@kelindi I'm trying not to change too much. I think it's about balance.

**Comment by kelindi** [comment]
@gmelodie I think it would add some more consistency when uploading a file, which would be better for dx. If I am understanding it correctly, the current route you have is the same no matter the file and then it figures out what the case is based on the file? If that is the case it would make more sense to have the request be formatted the same when uploading a regular file or a car file.

**Comment by gmelodie** [comment]
Closing in favor of #716

**Pull request closed without merging**